src/gui/layertree/qgslayertreeview.h

```diff
@@ -169,6 +169,29 @@ class GUI_EXPORT QgsLayerTreeView : public QTreeView
      */
     int layerMarkWidth() const { return mLayerMarkWidth; }
 
+    /**
+     * Returns the insertion group with its insertion index
+     * \since QGIS 3.10
+     */
+    QPair<QgsLayerTreeGroup *, int> insertionPoint() const SIP_SKIP;
```

Comment:
Can we avoid the pair and return a struct instead?

Reply:
actually I believe this relies in app in Qgis(App)Interface. It makes more sense to me since this logic can be app-dependent (with the logic for embedded layers for instance). I have another PR coming.

Reply:
Ok, fair enough